### PR TITLE
Added support for VAST ViewableImpression

### LINKED ISSUE #820: Support for VAST ViewableImpression
Currently, the player is unable to follow IAB's Viewable Impression events, as defined:

> Watching at least 2 seconds of the video where at least 50% of the ad’s pixels are visible on the screen

Implemention should be similar as other events, but view intersection needs to be calculated and current time during intersection must be tracked.

### PR DESCRIPTION
**Pull request by klaussilveira**
This PR adds support for IAB's Viewable Impression events, as defined:

> Watching at least 2 seconds of the video where at least 50% of the ad’s pixels are visible on the screen

It uses the IntersectionObserver API and follows the original impression implementation as closely as possible. Closes #820.

### CODE REVIEW

**Comment by klaussilveira** [comment]
@sergiomcsousa let me know if there is any interest in these. There's a couple more features from VAST/VPAID that I would like to implement.

**Comment by aron-ac** [comment]
Need this too 🙏

File: package-lock.json
Comment (ZentoDH) [inline package-lock.json:1]: I think the lock files can be removed

File: yarn.lock
Comment (ZentoDH) [inline yarn.lock:1]: Same here, you can delete the lock files

File: src/modules/utils.js [hunk 1]
```diff
@@ -183,6 +183,27 @@ export default function (playerInstance, options) {
         );
     }
 
+    playerInstance.observe = () => {
+        var observer = new IntersectionObserver(
```
Comment (ZentoDH) [inline src/modules/utils.js:187]: We would like to keep dependencies to a minimum. Ideally we would only want dependencies for stuff like VAST, Hls.js, ... 

Do you reckon you would be able to write a function like this in pure JS?

**Review by ZentoDH — COMMENTED** [review]
Looking good, thanks so much for your help! Just added one comment on there.

File: src/modules/utils.js (hunk 1 above)
Comment (klaussilveira) [reply src/modules/utils.js:187]: This API is widely available across a majority of browsers: https://caniuse.com/intersectionobserver

It's not really an external dependency.

Comment (ZentoDH) [reply src/modules/utils.js:187]: Ohhh haha sorry, stupid of me. In that case, just delete the lock files and I'll approve it.

**Review by Jhonsshons — COMMENTED** [review]
F

**Comment by ZentoDH** [comment]
@klaussilveira Just going to remove the lock files so we can move on in the process and QA can test it. If it passes, we'll probably release it this or next week.

**Comment by sbonetpi360** [comment]
Hi,
Thank you so much for this development. I wanted to ask if you know when this development will be available on npm. I see that on npm it is still in an older version v3.36.0 https://www.npmjs.com/package/fluid-player
Thanks,

**Comment by ZentoDH** [comment]
Hi @sbonetpi360 the npm package is now updated to the latest release. Thanks for bringing this to our attention!

**Comment by sbonetpi360** [comment]
@ZentoDH Thank you so much!

**Pull request merged**